### ISSUE TITLE
mail.app - your network preferences prevent content from loading privately

### ISSUE THREAD

**Issue by superandrew** [AdguardTeam/AdguardForMac#979]
### Steps to reproduce

1. Use Monterey
2. Enable ADGuard
3. In mail.app, open a mail with images or other content
4. "your network preferences prevent content from loading privately"
5. if you go to mail preferences and disable mail protection and "hide my IP" it works again
6. if you disable AdGuard, the mail displays content correctly

### Expected behavior
Adguard should disable whatever prevents it to work, or should work with mail protection feature in Monterey

### Actual behavior

You are forced to change settings on mail or disable adguard which is not ideal.

<details><summary>Screenshot:
![CleanShot 2021-10-31 at 11 04 58@2x](https://user-images.githubusercontent.com/71364/139577466-43ef9d48-e011-41ae-a736-3c9605bbd791.png)
</summary>

</details>

### Customer ID 

549115

### Your environment

* Environment name and version: (e.g. Chrome 59):
* Any specific potentially conflicting software installed: (e.g. antiviruses, firewalls, traffic counters, cleaners)

**Comment by Chinaski1** [comment]
Hello there!

I was able to reproduce the issue you mentioned.
We will try to find a solution to this issue in the near future.

As a temporary solution you can open Adguard -> Gear -> Advanced -> Advanced settings -> Change`network.extension.monterey.force.split.tunnel` to `true` value.

Thank you for the report.

**Comment by superandrew** [comment]
thank you @Chinaski1 . I will try your solution. Since I am no low level expert but I understand the general concepts, could you please elaborate what this setting does?

**Comment by demonbane** [comment]
I'd be curious to find out about this as well. I'm familiar with what a split tunnel is for a VPN, but some more details around what traffic is ignored by Adguard with this setting enabled would be helpful. (is it only Mail.app, all Monterey system services, all traffic over a certain port, etc?)

**Comment by sfionov** [comment]
@demonbane

AdGuard for Mac now uses macOS builtin socket filtering using "network extensions" API.
This is a new and buggy mechanism that replaced good old Kernel extensions. In last 1.5 years, we reported more than 20(!) bugs to Apple about their new filtering method.
This API is configured like VPN with list of route-like entries.

On Big Sur, AdGuard configured "split-tunnel" rules to not to create "default route" rule, because "default route" rule causes problems on early Big Sur releases.

On Monterey, iCloud Private Relay is introduced. Privacy features in Mail.app uses Private Relay servers too.

And now AdGuard can't work together with Private Relay and Mail.app privacy features. There's why:
1. Private Relay is applied to connection at library level, before connection reaches socket level, where AdGuard operates.
2. iCloud Private Relay uses QUIC which AdGuard can't filter in filtered apps, because HTTP/3 filtering is not implemented yet.
3. So AdGuard blocks QUIC, and, therefore iCloud Private Relay traffic, otherwise no ads will be blocked.
4. This means that when Private Relay and AdGuard in "split-tunnel" mode is on, sites will not be open in Safari.
5. To workaround this issue, on Monterey, we turned on "default route" rule. When Private Relay see that rule, it automatically disables itself. So, we don't break internet on Monterey, but Private Relay doesn't work too.

`network.extension.monterey.force.split.tunnel` just restores "Big Sur" behavior, but with this option access to sites may be broken due to (3) and (4).

We still searching the solution to this (including implementing HTTP/3 filtering).
But at this point we recommend to use more traditional VPN instead of new Apple's privacy features.
